x-pack/plugins/features/server/oss_features.ts

```diff
@@ -468,3 +483,118 @@ const timelionFeature: KibanaFeatureConfig = {
     },
   },
 };
+
+const reportingFeatures: {
+  [Property in
+    | 'generateCsvDiscover'
+    | 'downloadCsvDashboard'
+    | 'generateScreenshotDashboard'
+    | 'generateScreenshotVisualize']: SubFeatureConfig;
+} = {
+  generateCsvDiscover: {
+    name: i18n.translate('xpack.features.ossFeatures.reporting.discoverGenerateCSVFeatureName', {
+      defaultMessage: 'CSV Reports',
+    }),
+    privilegeGroups: [
+      {
+        groupType: 'independent',
+        privileges: [
+          {
+            id: 'generate_csv_report',
+            name: i18n.translate('xpack.features.ossFeatures.reporting.discoverGenerateCSV', {
+              defaultMessage: 'Generate CSV reports',
+            }),
+            includeIn: 'all',
```

Comment:
This is probably more of a product question (and perhaps you already had the discussion), but do we want reporting to be available to read-only users by default? It's currently restricted to users with the `All` privilege -- I could make an argument for either configuration, but I want to make sure that our decision is intentional/documented.

A gold license is required to customize sub-feature privileges, so this is primarily a concern for our basic users who don't have these options available to them.

Reply:
@legrego this is not intentional. As I was forging my way through examples, I saw this as the way to go. 

Let's chat about this more.

Reply:
The decision turned around pretty quickly - thanks for the input! 

> "granting reporting privileges to users with all privileges" means to use reporting, you need a role that grants "All" - or, if using a Gold+ license, you could go with a customized role that grants the specific privilege. 

It's intentional based on getting agreement from @elastic-jb